File: JavaScript/Algorithms/Stack Applications/palindrome expression.js
const isPalindrome = (expression) => {
    let len = expression.length, i = 0, stack = [];
    while(i < Math.floor(len/2)) {
        stack.push(expression[i++]);
    }

    // checks if expression length is even.
    i = len % 2 === 0? i: i+1;

    while(i < len) {
        if(stack.pop() != expression[i++]) {
            return false;
        }
    }
    return true;
}

console.log(isPalindrome("MbabM"));
console.log(isPalindrome("MaaM"));
console.log(isPalindrome('Muhamamad'));
console.log(isPalindrome('madam'));
console.log(isPalindrome('ababab'));
console.log(isPalindrome('aaaaa'));
console.log(isPalindrome('abababb'));
console.log(isPalindrome('ab'));
console.log(isPalindrome('a'));
console.log(isPalindrome('abababa'));

File: C#/Algorithms/Stack Applications/palindrome expression.cs
using System;
using System.Collections.Generic;

namespace palindrome_expression
{
    internal class Program
    {
        public static void Main(string[] args)
        {
            Console.WriteLine(IsPalindrome("madam"));
            Console.WriteLine(IsPalindrome("ababab"));
            Console.WriteLine(IsPalindrome("Muhammad"));
            Console.WriteLine(IsPalindrome("a"));
            Console.WriteLine(IsPalindrome("ab"));
            Console.WriteLine(IsPalindrome("aaaaaaaaa"));
            Console.WriteLine(IsPalindrome("azaza"));
        }

        public static bool IsPalindrome(string expression)
        {
            int length = expression.Length, i = 0;
            Stack<char> stack = new Stack<char>();

            while (i < length/2) 
            {
                stack.Push(expression[i++]);
            }

            i = length % 2 == 0 ? i : i + 1;

            while (i < length)
            {
                if (stack.Pop() != expression[i++])
                {
                    return false;
                }
            }
            return true;
        }
    }
}